include/eve/traits/product_type.hpp

```diff
@@ -149,7 +149,7 @@ namespace eve
     std::true_type derived_from_struct_support_impl(struct_support<Self, Fields...> const&);
 
     template <typename Type>
-    concept derived_from_struct_support = decltype(derived_from_struct_support_impl(Type{}))::value;
+    concept derived_from_struct_support = decltype(derived_from_struct_support_impl(std::declval<Type>()))::value;
```

Comment:
O_o? What's not default constructible

Reply:
it is not that it is not default constructible but we ended up trying to construct somethign with no constexpr constructor and clang was unhappy even if it made no sense.